core/src/system_monitor_service.rs

```diff
@@ -690,6 +695,112 @@ impl SystemMonitorService {
         })
     }
 
+    /*
+    fn string_append_u32_as_char(&mut string: String, val: u32)
+    {
+        let mut v = val;
+
+        for _i in 0..4 {
+            string.push(v as u8 as char);
+            v >>= 8;
+        };
+    }*/
+
+    #[cfg(any(target_arch = "x86_64", target_arch = "x86_64"))]
+    unsafe fn report_cpuid_values() {
+        let cpuid_0 = __cpuid(0);
+
+        let max_leaf = cpuid_0.eax;
+        let max_subleaf_7 = if 7 <= max_leaf {
+            __get_cpuid_max(7).1
+        } else {
+            0
+        };
+
+        let mut manufacturer = String::from("");
+        for i in 0..4 {
+            manufacturer.push((cpuid_0.ebx >> (i * 8)) as u8 as char);
+        }
+        for i in 0..4 {
+            manufacturer.push((cpuid_0.edx >> (i * 8)) as u8 as char);
+        }
+        for i in 0..4 {
+            manufacturer.push((cpuid_0.ecx >> (i * 8)) as u8 as char);
+        }
+
```

Comment:
assuming we keep the mfg string, let's avoid the bit twiddling and read these registers into a `Vec<u8>`, then attempt one string conversion at the end

something like...
```rust
let mut mfg = Vec::new();
mfg.extend_from_slice(&cpuid_0.ebx.to_le_bytes());
mfg.extend_from_slice(&cpuid_0.edx.to_le_bytes());
mfg.extend_from_slice(&cpuid_0.ecx.to_le_bytes());
let mfg = String::from_uftf(mfg).unwrap_or("dunno?".to_string());
```

Reply:
I went through a few versions of how to do this weird operation in Rust. Did not come up with that version!